MapleServer2/PacketHandlers/Game/UserChatHandler.cs

```diff
@@ -33,34 +35,58 @@ public override void Handle(GameSession session, PacketReader packet)
                 return;
             }
 
+            Packet itemLinkPacket = null;
+
+            // '<' signals a message containing an item link
+            if (message.Contains("<A"))
+            {
+                string[] itemLinkMessages = Regex.Matches(message, @"<A (.*?)>")
+                    .Cast<Match>()
+                    .Select(m => m.Value)
+                    .ToArray();
+                List<Item> item = new();
+
+                foreach (string itemLinkMessage in itemLinkMessages)
+                {
+                    string[] itemLinkMessageSplit = itemLinkMessage.Split(',');
+                    long itemUid = long.Parse(itemLinkMessageSplit[1]);
+
+                    item.Add(DatabaseManager.Items.FindByUid(itemUid));
+                    if (item != null)
```

Comment:
Item will always be not  null also change it to check for the count and you need to move it outside the for each.

Reply:
should be good now, thanks